app/api/api_v1/endpoints/data.py

```diff
@@ -17,54 +18,158 @@
 @router.post("/upload/", response_model=Message)
 async def upload(
         csv_file: UploadFile = File(...),
-        db: Session = Depends(deps.get_db)
+        db: Session = Depends(deps.get_db),
+        user: User = Depends(deps.get_current_user)
 ) -> Message:
     """
-    Upload a .csv file (Using the ploutos .csv provided as how a baseline file should be formatted)
+    Upload a .csv file that contains data about weather.
+
+    List of columns that this api will parse:
+
+    [
+        "date", "time", "parcel_location", "atmospheric_temperature", "atmospheric_temperature_daily_min",
+        "atmospheric_temperature_daily_max", "atmospheric_temperature_daily_average", "atmospheric_relative_humidity",
+        "atmospheric_pressure", "precipitation", "average_wind_speed", "wind_direction", "wind_gust",
+        "leaf_relative_humidity", "leaf_temperature", "leaf_wetness",
+        "soil_temperature_10cm", "soil_temperature_20cm", "soil_temperature_30cm", "soil_temperature_40cm",
+        "soil_temperature_50cm", "soil_temperature_60cm", "solar_irradiance_copernicus"
+    ]
+
+    date and time are mandatory fields.
+
+    date format: %Y-%m-%d
+    time format: %H:%M:%S
+
+    any decimal numbers present in the dataset should be formatted using "." (dot) and not "," (comma),
+    but nevertheless, the api will attempt to swap "," for ".".
+
     """
 
+    dataset_db = crud.dataset.get_by_name(db=db, name=csv_file.filename)
+
+    if dataset_db:
+        raise HTTPException(
+            status_code=400,
+            detail="Error, dataset with same filename already uploaded, please rename your dataset"
+        )
+
     csv_reader = reader(iterdecode(csv_file.file, "utf-8-sig"), delimiter=";")
-    first_column = True
+
+    # Parse the .csv headers
+    headers = next(csv_reader)
+
+    possible_column_names = utils.possible_column_names
+
+    # If there are more columns than what is expected, decline the dataset
+    if len(headers) > len(possible_column_names):
+        raise HTTPException(
+            status_code=400,
+            detail="Error, dataset has more than {} columns, not supported, please conform to required dataset format".format(len(possible_column_names))
+        )
+
+    # if there is no useful information then decline the dataset (see comment #A)
+    if len(headers) < 3:
+        raise HTTPException(
+            status_code=400,
+            detail="Error, won't accept dataset with less than 3 columns, please provide more information in the dataset"
+        )
+
+    if "date" not in headers or "time" not in headers:
```

Comment:
if both data & time are required why don't you use an `and` condition here?

Reply:
Switching "or" with "and" would cause the condition to change.
This would allow datasets like these:

date;atmo_pressure;atmo_temp
time;canopy_temp;leaf_wetness

to be accepted, which is bad, since these datasets are useless (if either date or time is missing, you lose key information about when a measurement was taken)

If by using an and condition you mean something like this:
if not ("date" in headers and "time" in headers)

then yes, it could be changed, but I think that may be a matter of taste more

Reply:
yes you are right